controllers/admin/AdminSaferPayOfficialPaymentController.php

```diff
@@ -212,17 +212,19 @@ public function getActiveCountriesList($onlyActive = true)
         return $countriesWithNames;
     }
 
-    public function getActiveCurrenciesList($onlyActive = true)
+    public function getActiveCurrenciesList(string $paymentMethod)
     {
-        $langId = $this->context->language->id;
-        $currencies = Currency::getCurrencies($langId, $onlyActive);
-        $currenciesWithNames = [];
-        $currenciesWithNames[0] = $this->l('All');
-        foreach ($currencies as $currency) {
-            $currenciesWithNames[$currency->id] = $currency->name;
+        /** @var \Invertus\SaferPay\Service\SaferPayObtainPaymentMethods $saferPayObtainPaymentMethods */
+        $saferPayObtainPaymentMethods = $this->module->getModuleContainer()->get(SaferPayObtainPaymentMethods::class);
+
+        $paymentMethods = $saferPayObtainPaymentMethods->obtainPaymentMethods();
+
+        $currencies[0] = $this->l('All');
+        foreach ($paymentMethods[$paymentMethod]['currencies'] as $currencyIso) {
+            $currencies[Currency::getIdByIsoCode($currencyIso)] = $currencyIso;
```

Comment:
Not sure if this will work for all cases as there won't be a currency in the shop like CHF for EUR based shops